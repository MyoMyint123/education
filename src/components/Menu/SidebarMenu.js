import React from 'react';
import MenuItem from './MenuItem';

const SidebarMenu = ({ menus }) => (
  <ul className="list-unstyled">
    <div>
      {/* <span>Logo will appear here</span> */}
    </div>
    {
      menus.map((menu) => <MenuItem key={menu.name} name={menu.name} linkTo={menu.link}  iconType={menu.iconType}/>)
    }
  </ul>
)

export default SidebarMenu;
