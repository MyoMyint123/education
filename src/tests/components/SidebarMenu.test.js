import React from 'react';
import { shallow } from 'enzyme';
import SidebarMenu from '../../components/Menu/SidebarMenu';
import menuItems from '../fixtures/menu-items';

describe('SidebarMenu', () => {
  let history;
  beforeEach(() => {
    history = { push: jest.fn() }
  });

  test('should render SidebarMenu correctly with items', () => {
    const wrapper = shallow(<SidebarMenu  menus={menuItems} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should render SidebarMenu without items', () => {
    const wrapper = shallow(<SidebarMenu menus={[]} />);
    expect(wrapper).toMatchSnapshot();
  });

});
