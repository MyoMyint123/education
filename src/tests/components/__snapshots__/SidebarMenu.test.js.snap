// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarMenu should render SidebarMenu correctly with items 1`] = `
<ul
  className="list-unstyled"
>
  <div />
  <MenuItem
    iconType="tachometer"
    key="Dashboard"
    linkTo="/dashboard"
    name="Dashboard"
  />
  <MenuItem
    iconType="graduation-cap"
    key="School"
    linkTo="/schools"
    name="School"
  />
  <MenuItem
    iconType="users"
    key="User"
    linkTo="/users"
    name="User"
  />
  <MenuItem
    iconType="cog"
    key="Setting"
    linkTo="/setting"
    name="Setting"
  />
</ul>
`;

exports[`SidebarMenu should render SidebarMenu without items 1`] = `
<ul
  className="list-unstyled"
>
  <div />
</ul>
`;
